File: android/core/designsystem/src/main/kotlin/com/nexussoft/verbum/designsystem/TokenGallery.kt
package com.nexussoft.verbum.designsystem

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.tooling.preview.Preview
import com.nexussoft.verbum.designsystem.tokens.Elevation
import com.nexussoft.verbum.designsystem.tokens.Radius
import com.nexussoft.verbum.designsystem.tokens.Spacing
import com.nexussoft.verbum.designsystem.tokens.VerbumTypography

// Visual reference for the token set. Open the preview to review typography,
// surfaces and accent in light/dark and at any font scale. Mirrors iOS TokenGallery.

@Preview(showBackground = true, name = "Tokens")
@Composable
private fun TokenGalleryPreview() {
    VerbumTheme {
        Column(
            Modifier.padding(Spacing.screenMargin),
            verticalArrangement = Arrangement.spacedBy(Spacing.xl),
        ) {
            Column(verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
                Text("David", style = VerbumTypography.editorialTitle)
                Text(
                    "King of Israel",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }

            Column(verticalArrangement = Arrangement.spacedBy(Spacing.md)) {
                Text("1 Samuel 17", style = VerbumTypography.scriptureHeading)
                Row(
                    horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
                    verticalAlignment = Alignment.Top,
                ) {
                    Text(
                        "45",
                        style = VerbumTypography.verseNumber,
                        color = MaterialTheme.colorScheme.outline,
                    )
                    Text(
                        "Then David said to the Philistine, “You come to me with a sword and with a spear and with a javelin, but I come to you in the name of the LORD of hosts.”",
                        style = VerbumTypography.scripture,
                    )
                }
            }

            Card(
                Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(Radius.lg),
                elevation = CardDefaults.cardElevation(defaultElevation = Elevation.Card.dp),
            ) {
                Column(Modifier.padding(Spacing.lg), verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
                    Text("Surface", style = MaterialTheme.typography.titleMedium)
                    Text(
                        "Surface container, card elevation.",
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                    )
                }
            }

            Button(onClick = {}) { Text("Explore connections") }
        }
    }
}

File: VerbumKit/Sources/DesignSystem/TokenGallery.swift
import SwiftUI

// Visual reference for the token set. Open the canvas on this file to review
// typography, surfaces and accent in light/dark and at any Dynamic Type size.
// Not shipped: previews are stripped from release builds.

#Preview("Tokens") {
    NavigationStack {
        ScrollView {
            VStack(alignment: .leading, spacing: Spacing.xl) {
                VStack(alignment: .leading, spacing: Spacing.sm) {
                    Text("David").font(Typography.editorialTitle)
                    Text("King of Israel")
                        .font(Typography.subheadline)
                        .foregroundStyle(Palette.foregroundSecondary)
                }

                VStack(alignment: .leading, spacing: Spacing.md) {
                    Text("1 Samuel 17").font(Typography.scriptureHeading)
                    HStack(alignment: .firstTextBaseline, spacing: Spacing.sm) {
                        Text("45")
                            .font(Typography.verseNumber)
                            .foregroundStyle(Palette.foregroundTertiary)
                        Text("Then David said to the Philistine, “You come to me with a sword and with a spear and with a javelin, but I come to you in the name of the LORD of hosts.”")
                            .font(Typography.scripture)
                            .lineSpacing(Typography.scriptureLineSpacing)
                    }
                }
                .padding(.horizontal, Spacing.readingMargin - Spacing.screenMargin)

                VStack(alignment: .leading, spacing: Spacing.sm) {
                    Text("Surface").font(Typography.headline)
                    Text("Secondary grouped background, card elevation.")
                        .font(Typography.footnote)
                        .foregroundStyle(Palette.foregroundSecondary)
                }
                .padding(Spacing.lg)
                .frame(maxWidth: .infinity, alignment: .leading)
                .background(Palette.surface, in: .rect(cornerRadius: Radius.lg))
                .elevation(.card)

                Button("Explore connections") {}
                    .buttonStyle(.borderedProminent)
                    .tint(Palette.accent)
            }
            .padding(Spacing.screenMargin)
        }
        .background(Palette.groupedBackground)
        .navigationTitle("Tokens")
    }
}
